programs/solana/programs/ics26-router/src/instructions/recv_packet.rs

```diff
@@ -0,0 +1,174 @@
+        mut,
+        seeds = [CLIENT_SEQUENCE_SEED, msg.packet.dest_client.as_bytes()],
+        bump
+    )]
+    pub client_sequence: Account<'info, ClientSequence>,
+
+    #[account(
+        init_if_needed,
+        payer = payer,
+        space = 8 + Commitment::INIT_SPACE,
+        seeds = [
+            PACKET_RECEIPT_SEED,
+            msg.packet.dest_client.as_bytes(),
+            &msg.packet.sequence.to_le_bytes()
+        ],
+        bump
+    )]
+    pub packet_receipt: Account<'info, Commitment>,
+
+    #[account(
+        init,
+        payer = payer,
+        space = 8 + Commitment::INIT_SPACE,
+        seeds = [
+            PACKET_ACK_SEED,
+            msg.packet.dest_client.as_bytes(),
+            &msg.packet.sequence.to_le_bytes()
+        ],
+        bump
+    )]
+    pub packet_ack: Account<'info, Commitment>,
+
+    pub relayer: Signer<'info>,
+
+    #[account(mut)]
+    pub payer: Signer<'info>,
+
+    pub system_program: Program<'info, System>,
+
+    pub clock: Sysvar<'info, Clock>,
+
+    // Client for light client lookup
+    #[account(
+        seeds = [CLIENT_SEED, msg.packet.dest_client.as_bytes()],
+        bump,
+        constraint = client.active @ RouterError::ClientNotActive,
+    )]
+    pub client: Account<'info, Client>,
+
+    // Light client verification accounts
+    /// CHECK: Light client program, validated against client registry
+    pub light_client_program: AccountInfo<'info>,
+
+    /// CHECK: Client state account, owned by light client program
+    pub client_state: AccountInfo<'info>,
+
+    /// CHECK: Consensus state account, owned by light client program
+    pub consensus_state: AccountInfo<'info>,
+}
+
+pub fn recv_packet(ctx: Context<RecvPacket>, msg: MsgRecvPacket) -> Result<()> {
+    // TODO: Support multi-payload packets #602
+    let router_state = &ctx.accounts.router_state;
+    let packet_receipt = &mut ctx.accounts.packet_receipt;
+    let packet_ack = &mut ctx.accounts.packet_ack;
+    let clock = &ctx.accounts.clock;
+
+    // Check authority (relayer must be authorized)
+    require!(
+        ctx.accounts.relayer.key() == router_state.authority,
+        RouterError::UnauthorizedSender
+    );
+
+    // Multi-payload check
+    require!(
+        msg.packet.payloads.len() == 1,
+        RouterError::MultiPayloadPacketNotSupported
+    );
+
+    // Validate timeout
+    let current_timestamp = clock.unix_timestamp;
+    require!(
+        msg.packet.timeout_timestamp > current_timestamp,
+        RouterError::InvalidTimeoutTimestamp
+    );
+
+    // Verify packet commitment on counterparty chain via light client
+    let client = &ctx.accounts.client;
+    let light_client_verification = LightClientVerification {
+        light_client_program: ctx.accounts.light_client_program.clone(),
+        client_state: ctx.accounts.client_state.clone(),
+        consensus_state: ctx.accounts.consensus_state.clone(),
+    };
+
+    let commitment_path = ics24::construct_commitment_path(
```

Comment:
use new paths, not ibc v1.

Reply:
oh, I guess they are used. Then yes, that needs to be fixed!